test/sets.jl

```diff
@@ -816,6 +816,22 @@ end
     @test_throws MethodError replace(identity, 1=>2, 3=>4)
     @test_throws MethodError replace!(identity, 1=>2)
     @test_throws MethodError replace!(identity, 1=>2, 3=>4)
+
+    # test replace and friends for AbstractDicts
+    d1 = GenericDict(Dict(1=>2, 3=>4))
+    d2 = replace(d1, (1=>2) => (1=>"a"))
+    @test d2 == Dict(1=>"a", 3=>4)
+    @test d2 isa Dict{Int, Any}
+    replace!(d1, (1=>2) => (1=>-2))
+    @test d1 == GenericDict(Dict(1=>-2, 3=>4))
```

Comment:
no need to wrap the rhs as a GenericDict
```suggestion
    @test d1 === replace!(d1, (1=>2) => (1=>-2))
    @test d1 == Dict(1=>-2, 3=>4)
```